File: src/my_ude_py/scripts/encoder.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Float32

RPM_DEFAULT_=5.0


class EncoderNode(Node): 
    def __init__(self):
        super().__init__("encoder") 
        self.declare_parameter("rpm_value", RPM_DEFAULT_)
        self.publisher_ = self.create_publisher(Float32, "rpm", 10)
        self.timer = self.create_timer(0.5, self.call_publish_rpm)
        self.get_logger().info("Encoder has been started")

    def call_publish_rpm(self):
         msg=Float32()
         self.rpm_ = self.get_parameter("rpm_value").get_parameter_value().double_value
         msg.data= self.rpm_
         self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = EncoderNode() 
    rclpy.spin(node)
    rclpy.shutdown()


if __name__ == "__main__":
    main()

File: src/my_ude_cpp/src/encoder.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/float32.hpp"

#include <chrono>

using namespace std::chrono_literals;

const double rpm_default_ = 5.0;


class EncoderNode : public rclcpp::Node 
{
public:
    EncoderNode() : Node("encoder") 
    {
        this->declare_parameter("rpm_value",rpm_default_);
        publisher_=this->create_publisher<std_msgs::msg::Float32>("rpm",10);
        timer_=this->create_wall_timer(1s, std::bind(&EncoderNode::callPublishRpm,this));
    	RCLCPP_INFO(this->get_logger(), "Encoder has been started.");
    }

private:

    void callPublishRpm(){
        auto msg=std_msgs::msg::Float32();
        rpm= this-> get_parameter("rpm_value").as_double();
        msg.data= rpm;
        publisher_->publish(msg);
    }

    float rpm;
    rclcpp::Publisher<std_msgs::msg::Float32>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;


};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<EncoderNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}